forge-gui-mobile/src/forge/adventure/stage/MapSprite.java

```diff
@@ -53,13 +50,7 @@ public static Array<Actor> GetMapSprites(int chunkX, int chunkY) {
 
         List<Pair<Vector2, Integer>> objects = WorldSave.getCurrentSave().getWorld().GetMapObjects(chunkX, chunkY);
 
-        for (Pair<Vector2, Integer> entry : objects) {
-            BiomeSpriteData data = WorldSave.getCurrentSave().getWorld().getObject(entry.getValue());
-            if (data.layer != SpriteLayer)
-                continue;
-            Actor sprite = new MapSprite(entry.getKey(), WorldSave.getCurrentSave().getWorld().getData().GetBiomeSprites().getSprite(data.name, (int) entry.getKey().x + (int) entry.getKey().y * 11483), null);
-            actorGroup.add(sprite);
-        }
+        actorGroup.addAll(GetMapSpritesBackground(chunkX, chunkY));
```

Comment:
spritelayer is different from background layer, why you remmove this? if you want to remove duplicate code, just update the method to differentiate spritelayer and backgroundlayer

Reply:
All of the calls are o(1), but there is admittedly the cost of the additional stack frames. I did verify that checkOut was unused.

Reply:
my bad the worldstage is not updated on master but was updated on snapshot version which it is used.. gonna commit it now
![image](https://user-images.githubusercontent.com/9781539/230695359-878e9fb3-6f6b-424a-a8f6-920b4f28a9f4.png)